ui/src/features/Editor/components/ParamsPanel/components/ParamEditor/index.tsx

```diff
@@ -1,132 +1,268 @@
 import { RJSFSchema } from "@rjsf/utils";
 import { JSONSchema7Definition } from "json-schema";
-import { memo, useMemo, useState } from "react";
+import { memo, useEffect, useMemo, useRef, useState } from "react";
 
-import { SchemaForm, Button } from "@flow/components";
+import {
+  SchemaForm,
+  Button,
+  Tabs,
+  TabsContent,
+  TabsTrigger,
+  TabsList,
+  FlowLogo,
+} from "@flow/components";
+import BasicBoiler from "@flow/components/BasicBoiler";
 import { patchAnyOfType } from "@flow/components/SchemaForm/patchSchemaTypes";
 import { useAction } from "@flow/lib/fetch";
 import { useT } from "@flow/lib/i18n";
 import i18n from "@flow/lib/i18n/i18n";
-import { batchNodeAction, noteNodeAction } from "@flow/lib/reactFlow";
+// import { generalNodeSchema } from "@flow/lib/reactFlow/nodeTypes/GeneralNode";
+import useNodeSchemaGenerate from "@flow/lib/reactFlow/useNodeSchemaGenerateHook";
 import type { NodeData } from "@flow/types";
 
 type Props = {
   nodeId: string;
   nodeMeta: NodeData;
   nodeType: string;
   nodeParameters?: unknown; // TODO: define type
-  onSubmit: (nodeId: string, data: any) => void;
+  onSubmit: (
+    nodeId: string,
+    data: any,
+    type: "params" | "customizations",
+  ) => Promise<void>;
 };
 
-// const actionButtonClasses = "border h-[25px]";
-
 const ParamEditor: React.FC<Props> = ({
   nodeId,
   nodeMeta,
-  // nodeType,
+  nodeType,
   // nodeParameters = [{ id: "param1", name: "Param 1", value: "Value 1", type: "string"}],
   onSubmit,
 }) => {
   const t = useT();
   const { useGetActionById } = useAction(i18n.language);
   let { action } = useGetActionById(nodeMeta.officialName);
+  const firstRenderRef = useRef(true);
+  // Used to generate the customization schema for the node with translations
+  const nodeSchema = useNodeSchemaGenerate(nodeType, nodeMeta.officialName);
 
+  // For nodes such as note and batch that are not in the actions list and therefore have no params.
```

Comment:
I feel like the following code should be in the `useNodeSchemaGenerate`. Currently that custom hook is just generation the schema for customization, which is weird.
Please consolidate so we are building the schema in one place 🙏🏻